File: src/bj9935/Main.java
package bj9935;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        Stack<Character> stack=new Stack<>();
        char[] input=br.readLine().toCharArray();
        int inputSize=input.length;
        char[] bomb=br.readLine().toCharArray();
        int bombSize=bomb.length;
        for(int i=inputSize-1; i>=0; i--){
            stack.push(input[i]);
            int stackSize=stack.size();
            if(stackSize>=bombSize){
                boolean isBomb=true;
                for(int j=stackSize-bombSize, bombIndex=bombSize-1; j<stackSize; j++, bombIndex--){
                    if(stack.get(j)!=bomb[bombIndex]){
                        isBomb=false;
                        break;
                    }
                }
                if(isBomb){
                    for(int j=0; j<bombSize; j++){
                        stack.pop();
                    }
                }
            }
        }
        if(stack.isEmpty())
            System.out.println("FRULA");
        else{
            int stackSize=stack.size();
            for(int i=0; i<stackSize; i++){
                System.out.print(stack.pop());
            }
        }
    }
}

File: kotlin/src/main/kotlin/bj9935/Main.kt
package bj9935

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val str = br.readLine()
    val bomb = br.readLine()
    val bombLastChar = bomb.last()
    val bombLength = bomb.length
    val afterBombStr = mutableListOf<Char>()
    for (i in str.indices) {
        afterBombStr.add(str[i])
        if (afterBombStr.last() == bombLastChar && afterBombStr.size >= bombLength) {
            var isBomb = true
            for (j in bomb.indices) {
                if (afterBombStr[afterBombStr.size - bombLength + j] != bomb[j]) {
                    isBomb = false
                    break
                }
            }
            if (isBomb) {
                for (j in 0 until bombLength) {
                    afterBombStr.removeLast()
                }
            }
        }
    }
    if (afterBombStr.isEmpty()) {
        println("FRULA")
    } else {
        println(afterBombStr.joinToString(""))
    }
}
